cmd/mailbox-controller/controller.go

```diff
@@ -143,7 +144,11 @@ func (ctl *mbCtl) enqueue(obj any) {
 		logger.V(4).Info("Enqueuing reference due to workspace", "wsName", typed.Name)
 		ctl.queue.Add(typed.Name)
 	case *edgev2alpha1.SyncTarget:
-		mbwsName := mbwsNameOfSynctarget(typed)
+		mbwsName := ctl.mbwsNameOfSynctarget(typed)
+		if mbwsName == "" {
+			logger.Error(nil, "Failed to construct mailbox workspace name from SyncTarget", "syncTargetName", typed.Name)
+			return
```

Comment:
This is not good enough handling. There needs to be a retry later. Find or create something that can be enqueued in this case.

Reply:
Changed to enqueue SyncTarget only on the recoverable failure and to retry later in the reconcile loop.

Reply:
done